File: collect_app/src/main/java/org/odk/collect/android/storage/ProjectStoragePaths.java
package org.odk.collect.android.storage;

import android.content.Context;

import java.io.File;

import dagger.assisted.Assisted;
import dagger.assisted.AssistedInject;

public class ProjectStoragePaths {

    private final String projectId;
    private final String externalFilesDir;

    @AssistedInject
    public ProjectStoragePaths(Context context, @Assisted String projectId) {
        this.externalFilesDir = context.getExternalFilesDir(null).getAbsolutePath();
        this.projectId = projectId;
    }

    public String getRootDir() {
        return externalFilesDir + File.separator + "projects" + File.separator + projectId;
    }
}

File: collect_app/src/main/java/org/odk/collect/android/storage/ProjectStoragePaths.kt
package org.odk.collect.android.storage

import android.content.Context
import dagger.assisted.Assisted
import dagger.assisted.AssistedInject
import java.io.File

class ProjectStoragePaths @AssistedInject constructor(
    context: Context,
    @Assisted private val projectId: String
) {
    private val externalFilesDir: String = context.getExternalFilesDir(null)!!.absolutePath

    val rootDir: String
        get() = externalFilesDir + File.separator + "projects" + File.separator + projectId
}
